File: examples/governance_static_go/domain/activate_agreement.go
package domain

import (
	"time"
	"fmt"
)

type ActivateAgreement struct {
	AgreementId string `json:"agreement_id"`
	EffectiveDate time.Time `json:"effective_date"`
	ExpirationDate time.Time `json:"expiration_date"`
}

func (c ActivateAgreement) CommandName() string { return "ActivateAgreement" }

func (c ActivateAgreement) Execute(repo DataUsageAgreementRepository) (*DataUsageAgreement, *ActivatedAgreement, error) {
	existing, err := repo.Find(c.AgreementId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("DataUsageAgreement not found: %s", c.AgreementId)
	}
	existing.EffectiveDate = c.EffectiveDate
	existing.ExpirationDate = c.ExpirationDate
	if existing.Status != "draft" {
		return nil, nil, fmt.Errorf("cannot ActivateAgreement: DataUsageAgreement is in %s state", existing.Status)
	}
	existing.Status = "active"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := ActivatedAgreement{
		AggregateID: existing.ID,
		AgreementId: c.AgreementId,
		EffectiveDate: c.EffectiveDate,
		ExpirationDate: c.ExpirationDate,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/model_registry_domain/lib/model_registry_domain/data_usage_agreement/commands/activate_agreement.rb
module ModelRegistryDomain
  class DataUsageAgreement
    module Commands
      class ActivateAgreement
        include Hecks::Command
        emits "ActivatedAgreement"

        attr_reader :agreement_id
        attr_reader :effective_date
        attr_reader :expiration_date

        def initialize(
          agreement_id: nil,
          effective_date: nil,
          expiration_date: nil
        )
          @agreement_id = agreement_id
          @effective_date = effective_date
          @expiration_date = expiration_date
        end

        def call
          existing = repository.find(agreement_id)
          if existing
            unless existing.status == "draft"
              raise ModelRegistryDomain::Error, "Cannot ActivateAgreement: status must be 'draft', got '#{existing.status}'"
            end
            DataUsageAgreement.new(
              id: existing.id,
              model_id: existing.model_id,
              data_source: existing.data_source,
              purpose: existing.purpose,
              consent_type: existing.consent_type,
              effective_date: effective_date,
              expiration_date: expiration_date,
              restrictions: existing.restrictions,
              status: "active"
            )
          else
            raise ModelRegistryDomain::Error, "DataUsageAgreement not found: #{agreement_id}"
          end
        end
      end
    end
  end
end
